docs/machine_learning.rst

```diff
@@ -89,30 +89,36 @@ Finally execute this whole training flow as a batch job::
     training_job = model.create_job()
     training_job.start_and_wait()
 
-
 Inference
 ----------
 
 When the batch job finishes successfully, the trained model can then be used
 with the ``predict_random_forest`` process on the raster data cube
 (or another cube with the same band structure) to classify all the pixels.
 
+Thus, the training job results is inspected to fetch the trained models url::
+
+    from openeo.rest.job import JobResults
+
+    results : JobResults = training_job.get_results()
+    links = results.get_metadata()['links']
+    ml_model_metadata_url = [link for link in links if 'ml_model_metadata.json' in link['href']][0]['href']
```

Comment:
Why getting the model URL from the links? Isn't this an asset in the batch job results?

Also, this "ml_model_metadata.json" looks like a VITO-specific implementation detail. Can this be generalized. Or at least documented why you are looking for that string?

Reply:
This is within the metadata indeed but different than the Assets's link. The link in the assets is as tar.gz . It throws a JSON error when trying to load it using that specific url and same is with the canonical link. 
ref: https://github.com/Open-EO/openeo-geopyspark-driver/issues/562

Reply:
If I understand  `save_ml_model` correctly (https://github.com/Open-EO/openeo-processes/pull/441/files#diff-0aae18f348c05c0f26c5caee2dcb0ca4af218a55287c9cbf8a286fc35b8faae5) there should be a asset that is a STAC Item that is to be used with `load_ml_model`.

that `'ml_model_metadata.json'` reference in the current docs is highly implementation specific and should be avoided

Reply:
I guess we are missing some specification to allow proper extraction of the model's URL, so I started the discussion at https://github.com/Open-EO/openeo-processes/pull/441/files/b162040dea04acd12fad73166274bbff370ed29b#r1650820880

Reply:
We need a better solution for 
```
[link for link in links if 'ml_model_metadata.json' in link['href']][0]['href']
```

this construct is quite brittle and implementation specific. 
Also note that https://github.com/Open-EO/openeo-processes/pull/441/files/b162040dea04acd12fad73166274bbff370ed29b#r1650872242 explicitly says that this the ML Model extension is outdated and going to be replaced, so it is very likely this is going to break

Reply:
This is apparently the next-gen ML extension
- https://github.com/crim-ca/mlm-extension

I think we better make sure we make this snippet future-proof